flink-examples/flink-examples-streaming/src/main/scala/org/apache/flink/streaming/scala/examples/ml/IncrementalLearningSkeleton.scala

```diff
@@ -62,19 +60,18 @@ object IncrementalLearningSkeleton {
 
     // set up the execution environment
     val env = StreamExecutionEnvironment.getExecutionEnvironment
-    env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
 
     // build new model on every second of new data
-    val trainingData: DataStream[Int] = env.addSource(new FiniteTrainingDataSource)
-    val newData: DataStream[Int] = env.addSource(new FiniteNewDataSource)
+    val trainingData: DataStream[Integer] = env.addSource(new FiniteTrainingDataSource)
```

Comment:
Why did we have to change to java classes here?

Reply:
The reason is that the explicit window assigners are usually a `WindowAssigner<Object, TimeWindow>`. This doesn't work for Scala `Int`, `Long`, etc. because they don't extend from `Object` but from `AnyVal`, so the Scala compiler will not allow it. It's not pretty, but I think the example is pretty artificial and in any real-world use case we would never have a `DataStream` of primitives.

Reply:
I see.